sklearn/decomposition/_nmf.py

```diff
@@ -1188,6 +1215,24 @@ def _check_params(self, X):
     def _check_w_h(self, X, W, H, update_H):
         """Check W and H, or initialize them."""
         n_samples, n_features = X.shape
+        if self.n_components == "auto":
+            if H is not None:
+                self._n_components = _num_samples(H)
+            elif W is not None:
+                self._n_components = _num_features(W)
+            else:
+                self._n_components = X.shape[1]
+
+        if W is not None and H is not None:
+            n_samples_H = _num_samples(H)
+            n_features_W = _num_features(W)
+            if n_samples_H != n_features_W:
+                raise ValueError(
+                    "Incompatible shapes for H and W. Expected n_features for"
+                    f" W and n_samples for H to be equal. Got {n_samples_H} and"
+                    f" {n_features_W} instead."
+                )
```

Comment:
So I think this check is not necessary anymore. the `_check_init` calls just below will complain if W and H don't have the appropriate shape.

Reply:
Deleted in b3529bb1a0b276990b2124907d9b2f9b8f790b5d